src/core/solvers/ddp.cpp

```diff
@@ -438,28 +447,61 @@ const std::vector<Eigen::VectorXd>& SolverDDP::get_k() const { return k_; }
 
 const std::vector<Eigen::VectorXd>& SolverDDP::get_fs() const { return fs_; }
 
-void SolverDDP::set_regfactor(const double& regfactor) {
+void SolverDDP::set_reg_incfactor(double regfactor) {
+  if (regfactor <= 1.) {
+    throw_pretty("Invalid argument: "
+                 << "reg_incfactor value is higher than 1.");
+  }
+  reg_incfactor_ = regfactor;
+}
+
+void SolverDDP::set_reg_decfactor(double regfactor) {
+  if (regfactor <= 1.) {
+    throw_pretty("Invalid argument: "
+                 << "reg_decfactor value is higher than 1.");
+  }
+  reg_decfactor_ = regfactor;
+}
+
+void SolverDDP::set_regfactor(double regfactor) {
```

Comment:
const

Reply:
Since this is deprecated, please call set_reg_decfactor and set_reg_incfactor from this function instead